### ISSUE TITLE
Can DISK be configured to perform causal (past-only) imputation?

### ISSUE THREAD

**Issue by HomnaA** [bozeklab/DISK#5]
Hi,

I'm using SLEAP to track the keypoints of rats performing a decision-making task, and DISK to impute missing keypoints due to occlusions or low-confidence detections.

For my downstream analysis, I use the animal's movement kinematics to predict its upcoming choice. Because of this, I would like to avoid any information leakage from future frames. Ideally, when imputing a missing keypoint at time t, I would like the model to use only observations from frames before t, rather than both past and future frames.

From the paper, my understanding is that DISK uses a Transformer with global self-attention (https://www.nature.com/articles/s41592-025-02893-y/figures/3), suggesting that the model has access to the entire input sequence (both before and after a missing segment) when performing imputation.

Is my understanding correct?

If so, is there currently any way to configure DISK to perform causal (past-only) imputation, for example, by using a causal attention mask or another setting? If not, do you have any recommendations for adapting DISK for this use case?

I'd really appreciate any guidance.

**Comment by HomnaA** [comment]
Hi,

Thank you so much for your reply. 
I have tried the causal version across several of my datasets, and so far it seems to perform comparably to the regular (acausal) DISK. For example, below is the RMSE during training/testing for one dataset, with regular DISK on the left and causal DISK on the right:

<img width="1610" height="321" alt="Image" src="https://github.com/user-attachments/assets/7bd7cbd8-a79b-4a4a-a3fd-6b90e6f533b8" />

I'm not sure whether the small difference in RMSE is meaningful, but I wanted to share the result.
